app/ui/browser/actions/main-actions.js

```diff
@@ -39,15 +39,19 @@ export function setStatusText(text) {
 }
 
 export function setUserTypedLocation(pageId, payload) {
-  return dispatch => {
+  return (dispatch) => {
     // Update this window's state before telling the profile service.
     dispatch({ type: types.SET_USER_TYPED_LOCATION, pageId, payload, instrument: false });
 
     // Only send request to profile service if there's a non-empty input.
     // Empty input could happen if a page finishes loading and the userTyped
     // state is going to be reset.
     if (payload.text) {
-      ipcRenderer.send('profile-command', profileCommands.setUserTypedLocation(payload.text));
+      profileCommands.request(profileCommands.setUserTypedLocation(payload.text))
+        .then(({ text, completionList }) => {
+          dispatch(profileDiffs.completions(text, completionList));
+        })
+        .catch(console.warn); // Ignore failures.  Not much to be done here.
```

Comment:
I don't think this will work how you want it to (although that's changing as of https://github.com/whatwg/console/issues/3 I don't think electron has caught up to it).  `console.warn.bind(console)` will do the trick